File: networklibrary/src/main/java/com/colantoni/federico/networklibrary/retrofit/adapters/wire/WireConverterAdapterRetrofitStrategyImpl.java
package com.colantoni.federico.networklibrary.retrofit.adapters.wire;


import com.colantoni.federico.networklibrary.retrofit.adapters.ConverterAdapterRetrofitStrategy;

import io.reactivex.annotations.Nullable;
import retrofit2.Converter;
import retrofit2.converter.wire.WireConverterFactory;


public class WireConverterAdapterRetrofitStrategyImpl implements ConverterAdapterRetrofitStrategy {

    @Override
    public Converter.Factory getConverterAdapterFactory(@Nullable final Object typeAdapter, final Object... typeAdapters) {

        return WireConverterFactory.create();
    }
}

File: networklibrary/src/main/java/com/colantoni/federico/networklibrary/retrofit/adapters/wire/WireConverterAdapterRetrofitStrategyImpl.kt
package com.colantoni.federico.networklibrary.retrofit.adapters.wire


import com.colantoni.federico.networklibrary.retrofit.adapters.ConverterAdapterRetrofitStrategy
import retrofit2.Converter
import retrofit2.converter.wire.WireConverterFactory


class WireConverterAdapterRetrofitStrategyImpl : ConverterAdapterRetrofitStrategy<Any> {

    override fun getConverterAdapterFactory(vararg typeAdapter: Any): Converter.Factory? {
        return WireConverterFactory.create()
    }
}
